simulation/app.py

```diff
@@ -0,0 +1,621 @@
+                        'infected': 'red',
+                        'recovered': 'green',
+                        'dead': 'gray'
+                     }, category_orders={'condition': ['susceptible', 'infected',
+                                                       'recovered', 'dead']})
+    fig.update_traces({
+        'marker': {'size': 10}
+    })
+    fig.update_layout({
+        'title': {'text': 'Population move', 'font': {'size': 30}},
+        'xaxis': {'title': 'Width'},
+        'yaxis': {'title': 'Height'},
+        'autosize': False,
+        'width': 800,
+        'height': 800,
+        'template': 'plotly_dark'
+    })
+
+    return fig
+
+
+@app.callback(Output('population-percent', 'figure'),
+              [Input('interval-component', 'n_intervals'),
+               Input('simulation-data', 'data')])
+def update_population_percent(n_intervals: int, json_data: json) -> 'plotly.express.pie':
+    """
+    Function handling pie chart with percent value of each group in population.
+
+    Parameters
+    ----------
+    n_intervals: int, required
+        Number of past intervals.
+    json_data: json, required
+        JSON contain data of simulation.
+        
+    Returns
+    -------
+    Plotly express object pie chart.
+    """
+
+    data = json.loads(json_data)
+    percent = {'condition': [], 'percent_value': []}
+
+    percent['condition'].append('susceptible')
+    percent['percent_value'].append(data['groups']['susceptible'])
+    percent['condition'].append('infected')
+    percent['percent_value'].append(data['groups']['infected'])
+    percent['condition'].append('recovered')
+    percent['percent_value'].append(data['groups']['recovered'])
+    percent['condition'].append('dead')
+    percent['percent_value'].append(data['groups']['dead'])
+
+    df = pd.DataFrame(data=percent)
+
+    fig = px.pie(df, values='percent_value', names='condition', color='condition',
+                 color_discrete_map={
+                        'susceptible': 'blue',
+                        'infected': 'red',
+                        'recovered': 'green',
+                        'dead': 'gray'
+                 })
+    fig.update_traces()
+    fig.update_layout({
+        'title': {'text': 'Percent of population in each group', 'font': {'size': 30}},
+        'template': 'plotly_dark'
+    })
+    return fig
+
+
+@app.callback(Output('current-population-amount', 'children'),
+              [Input('interval-component', 'n_intervals'),
+               Input('simulation-data', 'data')])
+def update_population_amount(n_intervals: int, json_data) -> 'html table':
+    """
+    Function handling display html table with amount of each group.
+
+    Parameters
+    ----------
+    n_intervals: int, required
+        Number of past intervals.
+    json_data: json, required
+        JSON contain data of simulation.
+
+    Returns
+    -------
+    Html table base od bash html.Table object.
+    """
+
+    data = json.loads(json_data)
```

Comment:
Why this is a binary data? Why not a normal python dict? Each time you need to load it and parse it. Is there any reason?